libs/core/src/lib/token/tokenizer.component.ts

```diff
@@ -19,54 +22,138 @@ import { TokenComponent } from './token.component';
+            this.previousTokenCount = this.tokenList.length;
+            this.handleTokenClickSubscriptions();
+        });
+        if (!this.compact) {
+            // because justify-content breaks scrollbar, it cannot be used on cozy screens, so use JS to scroll to the end
+            this.tokenizerInnerEl.nativeElement.scrollLeft = this.tokenizerInnerEl.nativeElement.scrollWidth -
+                this.tokenizerInnerEl.nativeElement.clientWidth;
+        }
+    }
+
+    /** @hidden */
+    handleTokenClickSubscriptions(): void {
+        this.tokenList.forEach((token, index) => {
+            token.onTokenClick.subscribe(event => {
+                this.focusTokenElement(event, index);
+            })
+        });
     }
 
     /** @hidden */
     handleKeyDown(event: KeyboardEvent, fromIndex: number): void {
         let newIndex: number;
         if (event.code === 'ArrowLeft') {
+            this.handleArrowLeft(fromIndex);
             newIndex = fromIndex - 1;
         } else if (event.code === 'ArrowRight') {
+            this.handleArrowRight(fromIndex);
             newIndex = fromIndex + 1;
         }
         if (newIndex === this.tokenList.length && event.code === 'ArrowRight') {
-            this.input.elementRef.nativeElement.focus();
+            this.input.elementRef().nativeElement.focus();
+        } else if (newIndex > this.tokenList.length - this.moreTokensRight.length &&
+            document.activeElement === this.input.elementRef().nativeElement) {
+            this.focusTokenElement(event, newIndex - this.moreTokensRight.length);
         } else if (newIndex || newIndex === 0) {
             this.focusTokenElement(event, newIndex);
         }
     }
 
+    private handleArrowLeft(fromIndex: number): void {
+        // if the leftmost visible token is selected, and there are moreTokensLeft, need to display a moreTokenLeft
+        if (fromIndex === this.moreTokensLeft.length) {
+            const poppedToken = this.moreTokensLeft.pop();
+            if (poppedToken) {
+                this.makeElementVisible(poppedToken.elementRef);
+            }
+            // and then hide any tokens from the right that no longer fit
+            this.collapseTokens('right');
+        }
+    }
+
+    private handleArrowRight(fromIndex: number): void {
+        if (fromIndex === this.tokenList.length - this.moreTokensRight.length - 1 && this.moreTokensRight.length) {
+            const poppedToken = this.moreTokensRight.pop();
+            if (poppedToken) {
+                this.makeElementVisible(poppedToken.elementRef);
+            }
+            // and then hide any tokens from the left that no longer fit
+            this.collapseTokens('left');
+        }
+    }
+
     /** @hidden */
-    focusTokenElement(event: KeyboardEvent, newIndex: number): HTMLElement {
+    focusTokenElement(event: Event, newIndex: number): HTMLElement {
```

Comment:
Order of methods rules should be followed, take a look [there](https://github.com/SAP/fundamental-ngx/wiki/Code-Contribution-Guideline)